### DISCUSSION TITLE
how can i register a java function in lua ?

### DISCUSSION THREAD

**Discussion by skinkairewalker** [gudzpoz/luajava#122] [Q&A] [answered]
hello everyone !
how can i register a java function with Register ?
Is it possible to register functions already created in the main class or inside other classes?

**Comment by gudzpoz** [comment, marked as answer]
I don't think there is a universal way to do this. But `push`ing the corresponding object, `setGlobal` to assign it to a global variable and wrapping it up with a Lua function should work:

```java
Function<String, String> f = (s) -> s + s;
L.push(f);
L.setGlobal("obj");
L.run("function f(s) obj:apply(s) end")
```
